flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/SubtaskMetricsHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.handler.legacy.metrics;
+
+import org.apache.flink.runtime.rest.handler.legacy.SubtaskExecutionAttemptDetailsHandler;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+/**
+ * Request handler that returns, aggregated across all subtasks of a single tasks, a list of all available metrics or the
+ * values for a set of metrics.
+ *
+ * <p>If the query parameters do not contain a "get" parameter the list of all metrics is returned.
+ * {@code {"available": [ { "name" : "X", "id" : "X" } ] } }
+ *
+ * <p>If the query parameters do contain a "get" parameter, a comma-separated list of metric names is expected as a value.
+ * {@code /metrics?get=X,Y}
+ * The handler will then return a list containing the values of the requested metrics.
+ * {@code [ { "id" : "X", "value" : "S" }, { "id" : "Y", "value" : "T" } ] }
+ *
+ * <p>The "agg" query parameter is used to define which aggregates should be calculated. Available aggregations are
+ * "sum", "max", "min" and "avg".
+ */
+public class SubtaskMetricsHandler extends AbstractMetricsHandler {
+	private static final String SUBTASK_METRICS_REST_PATH = "/jobs/:jobid/vertices/:vertexid/subtasks/:subtasknum/metrics";
+
+	public SubtaskMetricsHandler(Executor executor, MetricFetcher fetcher) {
+		super(executor, fetcher);
+	}
+
+	@Override
+	public String[] getPaths() {
+		return new String[]{SUBTASK_METRICS_REST_PATH};
+	}
+
+	@Override
+	protected Map<String, String> getMapFor(Map<String, String> pathParams, MetricStore metrics) {
+		String subtaskNumString = pathParams.get(SubtaskExecutionAttemptDetailsHandler.PARAMETER_SUBTASK_INDEX);
+		int subtaskNum;
+		try {
+			subtaskNum = Integer.valueOf(subtaskNumString);
+		} catch (NumberFormatException nfe) {
+			return null;
+		}
```

Comment:
Can this also be an invalid subtask number? E.g. we have 5 of them and we request subtaskid 11? In this case there should be a check that logs a warning and potentially sets he `subtaskNum` to `null` so that we go down the same path as a non-integer value.

Reply:
An unknown subtask index will cause the method to return null (```metrics.getSubtaskMetricStore(...)```), which is the same behavior as all other metric handlers.

We can't provide an accurate message in this case; we only know that this subtask index is unknown to the store, but not why. (does it exceed the parallelism, metrics haven't arrived yet).